.github/workflows/ci.yml

```diff
@@ -49,27 +63,39 @@ jobs:
         with:
           maven-version: 3.8.6
 
+      - name: Create cache dirs
+        run: mkdir -p ${{ env.FLINK_CACHE_DIR }}
+
+      - name: Cache Flink binary
+        if: ${{ inputs.cache_flink_binary == 'true' }}
+        uses: actions/cache@v3
+        id: cache-flink
+        with:
+          path: ${{ env.FLINK_CACHE_DIR }}
+          key: ${{ inputs.flink_url }}
+
       - name: Download Flink binary
-        working-directory: ./..
-        run: wget -q -c ${{ env.FLINK_URL }} -O - | tar -xz
+        working-directory: ${{ env.FLINK_CACHE_DIR }}
+        if: steps.cache-flink.outputs.cache-hit != 'true'
+        run: wget -q -c ${{ inputs.flink_url }} -O - | tar -xz
 
-      - name: Compile and test flink-connector-elasticsearch
-        timeout-minutes: 20
-        run: |
+      - name: Compile and test flink-connector-rabbitmq
+        timeout-minutes: 30
+        run: |          
           set -o pipefail
 
           mvn clean deploy ${MVN_COMMON_OPTIONS} \
             -DaltDeploymentRepository=validation_repository::default::file:${{ env.MVN_VALIDATION_DIR }} \
             -Dscala-2.12 \
-            -Prun-end-to-end-tests -DdistDir=$(pwd)/../flink-1.16-SNAPSHOT \
+            -Prun-end-to-end-tests -DdistDir=${{ env.FLINK_CACHE_DIR }}/flink-${{ inputs.flink_version }} \
```

Comment:
will you merge this commit into the previous one? Most of these changes shouldn't be necessary.